### PR TITLE
Fix all problems on @phpstan level 5

### PR DESCRIPTION
**Pull request by szepeviktor**
@barryvdh Please consider going on: `phpstan analyze src/ -l max`

### CODE REVIEW

**Comment by szepeviktor** [comment]
travis-ci.com runs much better, and travis-ci.org will be shut down.

File: src/Common/CreditCard.php [hunk 1]
```diff
@@ -255,7 +255,7 @@ public function validate()
             throw new InvalidCreditCardException('Card number is invalid');
         }
 
-        if (!is_null($this->getNumber()) && !preg_match('/^\d{12,19}$/i', $this->getNumber())) {
+        if ($this->getNumber() !== '' && !preg_match('/^\d{12,19}$/i', $this->getNumber())) {
```
Comment (barryvdh) [inline src/Common/CreditCard.php:258]: Is this always a string? Cause the check is changed here. Perhaps the phpdocs for the getNumber are not correct instead?

Comment (szepeviktor) [reply src/Common/CreditCard.php:258]: I really do not know, I'm not a user and this is an inspirational PR to promote PHPStan :)

**Comment by barryvdh** [comment]
I'm closing this because I'm not really sure of the changes.

**Comment by szepeviktor** [comment]
Recruitment of Barry failed 😿

**Comment by barryvdh** [comment]
I'm all in favor of phpstan, but not blindly doing what it suggests.

**Pull request closed without merging**